darts/timeseries.py

```diff
@@ -2849,7 +2854,10 @@ def append(self, other: Self) -> Self:
             "Both series must have the same number of components.",
             logger,
         )
-        if self._has_datetime_index:
+        # In the Rangeindex we never call this (especially when empty), thus not resulting in the
+        # bug when ultimately calling "other.start_time()"
+        # Checking that other.time_index is NOT EMPTY and only then entering should fix the issue
+        if self._has_datetime_index and (not other.time_index.empty):
```

Comment:
Thinking about, it might be better to also check this for the non-datetime index (range index). This will introduce a breaking change (which we need to highlight in the CHANGELOG.md) but at least we're consistent with our notion of time across all index types.

Also this check requires `self` not to be empty either

```suggestion
        if len(self) > 0 and len(other) > 0:
```